### ISSUE TITLE
Unformatted devices cannot be resized

### ISSUE THREAD

**Issue by vojtechtrefny** [storaged-project/blivet#787]
Device is not resizable if the format is not resizable and the "default" `DeviceFormat` is not resizable.

**Comment by vpodzime** [comment]
Isn't this on purpose to prevent issues with resizing devices with unknown formats?

**Comment by vojtechtrefny** [comment]
Yes, this would require bigger changes in the `DeviceFormat`, probably creating some new "UnknownFormat" class for devices with some format (recognized by blkid) but without support in blivet, and maybe even some special "NoFormat" for devices without a format (or format not recognized by blkid, but I think we can safely say device is not formatted if blkid tells us there is nothing on it).

**Comment by dwlehman** [comment]
There is also this to differentiate between "blkid reported nothing" and "blivet chose DeviceFormat":

  https://github.com/storaged-project/blivet/blob/3.1-devel/blivet/formats/__init__.py#L105

**Comment by dwlehman** [comment]
An unformatted device could also be removed/replaced as a workaround. This all to avoid the case of an angry careless person whose device containing some rare formatting was resized, causing data loss.
